pennylane_lightning_gpu/src/algorithms/ObservablesGPU.hpp

```diff
@@ -446,12 +446,16 @@ class SparseHamiltonianGPU final : public ObservableGPU<T> {
                                         std::move(arg3), std::move(arg4)});
     }
 
-    // to work with
+    /**
+     * @brief Updates the statevector SV:->SV', where SV' = a*H*SV, and where H
+     * is a sparse Hamiltonian.
+     *
+     */
     void applyInPlace(StateVectorCudaManaged<T> &sv) const override {
```

Comment:
I think it is better to check if the dimension of statevector matches with the matrix shape.

Reply:
Yep, great point. I'll add a safety check for this.

Reply:
Since the `qml.SparseHamiltonian` [docs](https://docs.pennylane.ai/en/stable/code/api/pennylane.SparseHamiltonian.html) state the SpH should be used over the entire wire-space of the device, I'll use the wires as a proxy to check the size here.